src/frontend/src/components/base/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,106 @@
+import { useAsyncList, type AsyncListData } from "@react-stately/data";
+import {
+    Autocomplete as AriaAutocomplete,
+    Input,
+    Menu,
+    MenuItem,
+    SearchField,
+} from "react-aria-components";
+
+import TimesIcon from "assets/icons/times.svg?react";
+import Button from "components/base/Button/Button";
+import {
+    dropdownItemStyles,
+    dropdownPopoverStyles,
+} from "../Dropdown/Dropdown.styles";
+import autocompleteStyles from "./Autocomplete.styles";
+
+type Suggestion = { name: string; [key: string]: any };
+
+export function Autocomplete({
+    placeholder,
+    suggestions,
+    loadAsyncSuggestions,
+    onSuggestionCallback,
+    onClearCallback,
+}: {
+    placeholder: string;
+    suggestions?: Suggestion[];
+    loadAsyncSuggestions?: (
+        query: string,
+        signal: AbortSignal
+    ) => Promise<Suggestion[]>;
+    onSuggestionCallback?: (suggestion: Suggestion) => void;
+    onClearCallback?: () => void;
+}) {
+    const { searchRoot, input: inputStyles } = autocompleteStyles();
+    const baseDropdownPopoverStyles = dropdownPopoverStyles();
+    const baseDropdownItemStyles = dropdownItemStyles();
+
+    let filteredSuggestions: AsyncListData<Suggestion> = useAsyncList({
```

Comment:
This is an interesting first-use of `useAsyncList` in this project 👏🏼. I'd suggest adding some comments on what it does!